### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by lewiskg**
Part4 done.

### CODE REVIEW

File: javascripts/controllers/NewCtrl.js [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
+		// console.log("Hello, NewCtrl. Route:/contacts/edit/:id");
+	} else {
+		$scope.controller = "Hello, NewCtrl. Route:/contacts/new";
+		// console.log("Hello, NewCtrl. Route:/contacts/new");
+	}
 
 	$scope.saveContact = (contact) => {
 		contact.uid = $rootScope.uid;
-		// console.log(contact);			
+		contact.rating = 0;
```
Comment (morecallan) [inline javascripts/controllers/NewCtrl.js]: Ken! Why are you rating your friends?? (Rating isn't a key you have in your json or database).

File: javascripts/controllers/NewCtrl.js [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
-	// 		"phoneHome": contact.phoneHome,
-	// 		"phoneWork": contact.phoneWork,
-	// 		"emailHome": contact.emailHome,
-	// 		"emailWork": contact.emailWork,
-	// 		"notes":	 contact.notes,
-	// 		"uid": $rootScope.uid
-	// 	};
-	// };
+	if($rootScope.flag) {
+		$scope.controller = "Hello, NewCtrl. Route:/contacts/edit/:id";
```
Comment (morecallan) [inline javascripts/controllers/NewCtrl.js]: At this point, you don't need to have these "hello" messages print in the partials. Clever way of doing this, though!

**Review by morecallan — COMMENTED** [review]
Hey @lewiskg - looking really good. Strong form validation. But 2 bugs. 1) Can't delete the last contact 2) When I go to edit, if I don't fill out anything in the form, then it won't let me save to go back the the main view.

File: javascripts/controllers/NewCtrl.js (hunk 1 above)
Comment (lewiskg) [reply javascripts/controllers/NewCtrl.js]: Fixed

File: javascripts/controllers/NewCtrl.js (hunk 2 above)
Comment (lewiskg) [reply javascripts/controllers/NewCtrl.js]: Fixed

**Comment by lewiskg** [comment]
Haven't pushed up corrections yet.

**Comment by lewiskg** [comment]
Part4 fixes pushed up. Done.

**Comment by lewiskg** [comment]
Note: Some how I introduced a bug into the "put request" in the updated PR after Callan reviewed the original PR.  I didn't notice until working on part5.  It has been corrected in the part5 PR.

**Comment by zoeames** [comment]
looks good!

**Pull request merged**